src/pip/_internal/req/constructors.py

```diff
@@ -145,7 +145,7 @@ def deduce_helpful_msg(req):
         msg = " The path does exist. "
         # Try to parse and check if it is a requirements file.
         try:
-            with open(req, 'r') as fp:
+            with open(req) as fp:
```

Comment:
This may fail on Windows if non-ASCII comments in the requirements file.
Please add `encoding="utf-8"`.

Reply:
Similarly, the encoding parameter would be a backward-incompatible change, because historically pip has been reading the file with platform encoding. We can (and probably should at some point) switch to UTF-8 everywhere, but that needs to go through a deprecation cycle, and should be treated as a separate feature request.